### ISSUE TITLE
Subfolder in res.zip of StarDict dictionaries

### ISSUE THREAD

**Issue by mokkiii** [goldendict/goldendict#621]
Resources of StarDict dictionaries don't seem to be loaded if the `res.zip` archive itself contains a `res` folder with all the resources in. I was only able to get it working when the zip archive had all the resources in the zip's root directory.

I tested this on the pre-compiled OS X version from the forums: http://goldendict.org/forum/viewtopic.php?f=4&t=1933&start=60

**Comment by Abs62** [comment]
Yes, res.zip file should not contain any folders. It is not bug.

**Comment by ghostly123** [comment]
The thing is that it's extremely difficult (if not impossible) to create a zip archive consisting of ten thousands of files if they are not contained inside a folder. For example `zip` in OS X's Terminal quits with `Argument list too long` and I could imagine that selecting such an amount of files in any file browser might cause extreme lags and even crashs, too. Zipping a single folder that contains the files in question is definitely easier and the usual way of creating archives.

So in addition to this

```text
res.zip
|-- file1
|-- file2
|-- file3
|-- ...
```

it should be possible to have GoldenDict recognize archives like this

```text
res.zip
|-- res
    |-- file1
    |-- file2
    |-- file3
    |-- ...
```

**Comment by eugenesvk** [comment]
I would second @ghost suggestion. My dictionaries also came with a `res` folder within `res.zip` and it's indeed rather difficult to create a zip of tens of thousans of files (what a stupid format that requires every single word to have a single .jpg/.wav (and .wav of all formats!) file)
